{
	"documenttitle": {
		"text": "Red Hat Security Advisory: python-setuptools security update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2023:0952"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2023-02-28T08:24:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2023-02-28T08:24:00Z",
		"CurrentReleaseDate": "2023-02-28T08:24:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.11.0",
			"Date": "2023-02-28T12:46:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update for python-setuptools is now available for Red Hat Enterprise Linux 9.\n\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nThe python-setuptools package provides a collection of enhancements to Python distribution utilities allowing convenient building and distribution of Python packages.\n\nSecurity Fix(es):\n\n* pypa-setuptools: Regular Expression Denial of Service (ReDoS) in package_index.py (CVE-2022-40897)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2023 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2023:0952",
				"Description": "https://access.redhat.com/errata/RHSA-2023:0952"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A flaw was found in Python Setuptools due to a regular expression Denial of Service (ReDoS) present in package_index.py. This issue could allow a remote attacker to cause a denial of service via HTML in a crafted package or custom PackageIndex page.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2022-12-22T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2022-40897",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2023:0952"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://pyup.io/vulnerabilities/CVE-2022-40897/52495/",
						"Description": "https://pyup.io/vulnerabilities/CVE-2022-40897/52495/"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2022-40897",
						"Description": "CVE-2022-40897"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2158559",
						"Description": "bz#2158559: CVE-2022-40897 pypa-setuptools: Regular Expression Denial of Service (ReDoS) in package_index.py"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2023-01-05T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"BaseOS-9.1.0.Z.MAIN:python-setuptools-0:53.0.0-10.el9_1.1"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Enterprise Linux",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux BaseOS (v. 9)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux BaseOS (v. 9)",
							"productid": "BaseOS-9.1.0.Z.MAIN",
							"cpe": "cpe:/o:redhat:enterprise_linux:9::baseos"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "python-setuptools-0:53.0.0-10.el9_1.1",
				"fullproductname": {
					"text": "python-setuptools-53.0.0-10.el9_1.1.src.rpm",
					"productid": "python-setuptools-0:53.0.0-10.el9_1.1",
					"cpe": "cpe:/o:redhat:enterprise_linux:9::baseos"
				}
			}
		],
		"relationship": [
			{
				"productreference": "python-setuptools-0:53.0.0-10.el9_1.1",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "BaseOS-9.1.0.Z.MAIN",
				"fullproductname": {
					"text": "python-setuptools-0:53.0.0-10.el9_1.1 as a component of Red Hat Enterprise Linux BaseOS (v. 9)",
					"productid": "BaseOS-9.1.0.Z.MAIN:python-setuptools-0:53.0.0-10.el9_1.1",
					"cpe": "cpe:/o:redhat:enterprise_linux:9::baseos"
				}
			}
		]
	}
}
